python/ray/tune/search/sample.py

```diff
@@ -571,7 +578,11 @@ def quniform(lower: float, upper: float, q: float):
     Quantization makes the upper bound inclusive.
 
     """
-    return Float(lower, upper).uniform().quantized(q)
+    return (
+        Float(lower, upper).uniform().quantized(q)
+        if q != 1.0
```

Comment:
just curious, why do we need to special-case here?

Is `quantized()` not expecting `q = 1`?

Reply:
Agreed, we can remove the special cases

Reply:
By adding the special case, we can remove unnecessary quantization logic that would be called otherwise. Also the quantization algorithm does not make the upperbound exclusive when `q = 1`

Reply:
Can we have this case inside the quantization itself then?